src/applications/personalization/profile-2/components/MilitaryInformation.jsx

```diff
@@ -1,5 +1,182 @@
+import AlertBox from '@department-of-veterans-affairs/formation-react/AlertBox';
 
-export default MilitaryInformation;
+import recordEvent from 'platform/monitoring/record-event';
+import facilityLocator from 'applications/facility-locator/manifest.json';
+
+class MilitaryInformationContent extends React.Component {
+  componentDidMount() {
+    this.props.fetchMilitaryInformation();
+  }
+
+  renderContent = () => {
+    const {
+      serviceHistory: { serviceHistory, error },
+    } = this.props.militaryInformation;
+
+    if (error) {
+      if (some(error.errors, ['code', '403'])) {
+        return (
+          <AlertBox
+            isVisible
+            status="warning"
+            headline="We can’t access your military information"
+            content={
+              <div>
+                <p>
+                  We’re sorry. We can’t find your Department of Defense (DoD)
+                  ID. We need this to access your military service records.
+                  Please call us at 800-827-1000, or visit your nearest VA
+                  regional office and request to be added to the Defense
+                  Enrollment Eligibility Reporting System (DEERS).
+                </p>
+                <a href={facilityLocator.rootUrl}>
+                  Find your nearest VA regional office
+                </a>
+                .
+                <p>
+                  You can also request to be added to DEERS through our online
+                  customer help center.
+                </p>
+                <a href="https://iris.custhelp.va.gov/app/answers/detail/a_id/3036/~/not-registered-in-deers%2C-or-received-and-error-message-while-trying-to">
+                  Get instructions from our help center
+                </a>
+                .
+              </div>
+            }
+          />
+        );
+      } else if (some(error.errors, e => ['500', '503'].includes(e.code))) {
+        return <LoadFail information="military" />;
+      }
+    }
+
+    if (serviceHistory.length === 0) {
+      return (
+        <AlertBox
+          isVisible
+          status="warning"
+          headline="We can’t access your military information"
+          content={
+            <p>
+              We’re sorry. We can’t access your military service records. If you
+              think you should be able to view your service information here,
+              please file a request to change or correct your DD214 or other
+              military records.
+            </p>
+          }
+        />
+      );
+    }
+
+    return (
+      <table className="militaryInformation" data-field-name="serviceHistory">
+        <thead>
+          <tr>
+            <th colSpan="2">
+              <h3>Period of Service</h3>
+            </th>
+          </tr>
+        </thead>
+        <tbody>
+          {serviceHistory.map((service, index) => (
+            <tr key={index}>
+              <td>
+                <span>
```

Comment:
just curious, is there a reason you went with two spans in this single table cell vs. having two cells in the row?

Reply:
@erikphansen I was having issues getting everything lined up properly - but I can go back and try using ```<td>```

Reply:
That just feels more semantic (maybe the wrong word). I guess I don't have super strong feelings either way. And I haven't worked with tables in forever so it might be a nightmare.

Reply:
@erikphansen It wasn't particularly fun trying to collapse the 2 <td> and retain the outside border but not border between cells